alibi/explainers/backends/pytorch/cfrl_tabular.py

```diff
@@ -76,9 +76,11 @@ def l0_ohe(input: torch.Tensor,
 
     Returns
     -------
-    L0 loss.
+        L0 loss.
     """
-    loss = torch.maximum(target - input, torch.zeros_like(input))
+    # Order matters as the gradient of zeros will still flow if reversed order. Maybe consider clipping a bit higher?
+    eps = 1e-7 / input.shape[1]
+    loss = torch.maximum(target - input, eps + torch.zeros_like(input)).sum(dim=1)
```

Comment:
What's the reason for having to add `eps`?

Reply:
@RobertSamoilescu followed-up on this privately: essentially `torch` and `tensorflow` have a different behaviour on which element of `maximum` is picked if they happen to be equal. This has an effect on gradient flow which is a side-effect we depend on here. The solution here is to add a small `eps` so that the order does not play a role anymore and we can rely on stopping the gradient flow for either backend.